### PR TITLE
[PropertyInfo] throw exception if docblock factory does not exist

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #26259
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php [hunk 1]
```diff
@@ -53,6 +53,10 @@ class PhpDocExtractor implements PropertyDescriptionExtractorInterface, Property
      */
     public function __construct(DocBlockFactoryInterface $docBlockFactory = null, array $mutatorPrefixes = null, array $accessorPrefixes = null, array $arrayMutatorPrefixes = null)
     {
+        if (!class_exists(DocBlockFactoryInterface::class)) {
```
Comment (hkdobrev) [inline src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php]: Shouldn't this be `interface_exists`?

http://php.net/manual/en/function.interface-exists.php

**Review by hkdobrev — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php]: good catch, in fact I wanted to use the `DocBlockFactory` class here as that's what is used below

**Comment by hkdobrev** [comment]
@xabbuh 👍 I guess this is indeed not a BC break, but this doesn't solve the side effect I've described in https://github.com/symfony/symfony/issues/26259#issuecomment-367465746

When you have the phpDocumentor installed locally, because of another dependency like PHPUnit, this would not throw an exception locally or in testing, but it would throw an exception on production without #26260.

**Comment by xabbuh** [comment]
That's true, but I don't think we can cover that in Symfony itself.

**Comment by hkdobrev** [comment]
@xabbuh OK. What's your understanding of how that should be handled in the projects using the component? You should require a library noted in the docs even though you're not using its code directly, but it's used from your dependency?

Shouldn't the `PhpDocExtractor` be extracted into a separate package and then that package would have the phpDocumentor in its `require` block? Like that if you use the `PhpDocExtractor` class you need to install its package and it would work without knowing what are its dependencies. Easier to document as well.

File: src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php [hunk 2]
```diff
@@ -53,6 +53,10 @@ class PhpDocExtractor implements PropertyDescriptionExtractorInterface, Property
      */
     public function __construct(DocBlockFactoryInterface $docBlockFactory = null, array $mutatorPrefixes = null, array $accessorPrefixes = null, array $arrayMutatorPrefixes = null)
     {
+        if (!class_exists(DocBlockFactory::class)) {
+            throw new \RuntimeException(sprintf('Unable to use the %s class as the phpdocumentor/reflection-docblock package is not installed.', __CLASS__));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php]: missing double quotes

**Review by hkdobrev — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**